### ISSUE TITLE
Version should support tags such as "rc", "beta", etc.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3086]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)| @radix reported|
|-|-|
|Trac ID|trac#3086|
|Type|enhancement|
|Created|2008-03-17 17:09:13Z|
|Branch|https://github.com/twisted/twisted/tree/version-identifier-3086|

We need to release 8.0.0beta1.

In addition, the release version updating code should allow you to specify it.

<details><summary>Searchable metadata</summary>

```
trac-id__3086 3086
type__enhancement enhancement
reporter__radix radix
priority__normal normal
milestone__twisted_8_0 twisted-8.0
branch__branches_version_identifier_3086 branches/version-identifier-3086
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1205773753000000 1205773753000000
changetime__1205799335000000 1205799335000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_1"></a>|@radix commented|
|-|-|

(In [22886]) Branching to 'version-identifier-3086'

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_2"></a>|@radix set owner to @exarkun|
|-|-|

I decided to not implement arbitrary stringy things, but instead just a simple prerelease= parameter.

Up for review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @radix|
|-|-|

Okay, looks good, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

Don't really merge.  Do the update stuff.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_5"></a>|@radix set owner to @exarkun|
|-|-|

Ok, ready for review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set owner to @radix|
|-|-|

  * `Version.__str__` should call `Version.base`, shouldn't it?
  * `getNextVersion` doesn't really make sense for the prerelease case, eg that `test_getNextVersionWithPrerelease` is testing.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_7"></a>|@radix set status to `closed`|
|-|-|

(In [22909]) Merge version-identifier-3086

Author: radix
Reviewer: exarkun
Fixes [#3086](https://github.com/twisted/twisted/issues/3086)

Versions can now represent prereleases and the release code now allows
you to write version files with prereleases.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_8"></a>|@radix set status to `reopened`|
|-|-|

(In [22912]) Revert r22909: buildbot failures

There's a failure on 2.3-based buildslaves, basically because inf > inf.
It's unknown why it even works on 2.4 and 2.5.

Reopens [#3086](https://github.com/twisted/twisted/issues/3086)

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_9"></a>|@radix set owner to @therve<br>@radix set status to `new`|
|-|-|

ready for review again. I've made _inf == _inf.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_10"></a>|@therve set owner to @radix|
|-|-|

 * typo in test_replaceProjectVersionWithPrerelease docstring
 * please add a docstring to test_versionComparison
 * missing tab in last line of __cmp__ docstring

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_11"></a>|@radix set status to `closed`|
|-|-|

(In [22922]) Merge version-identifier-3086

Author: radix
Reviewer: exarkun
Fixes [#3086](https://github.com/twisted/twisted/issues/3086)

Versions can now represent prereleases and the release code now allows
you to write version files with prereleases.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
